### DISCUSSION TITLE
Usage of BetterPlayerController on a local server

### DISCUSSION THREAD

**Discussion by ferangmor** [jhomlala/betterplayer#483] [Q&A]
I have an odoo 11 server from which i want to send a video that i store in a database. Users should have access to their videos using a flutter app. 
I have created a secured endpoint with jwt. which is important bc the videos are private for each user. This endpoint sends a response type video/mp4.
This is a capture of the Postman call: [Gyazo link](https://gyazo.com/88ce87f14c8245fe160251da8f64c32c)
`
Map<String,String> mapa = {"Authorization": widget.jwtToken,
      "Connection":"keep-alive",
      "Range":"bytes=0-"};

BetterPlayerDataSource betterPlayerDataSource = BetterPlayerDataSource(
      BetterPlayerDataSourceType.network,
      "http://localhost:8069/downloadVideoFile/21/4",
      headers: mapa
      );`

The thing is that i don even know if this shoukd work, i am getting a ExoPlayer error:
Unhandled Exception: PlatformException(VideoError, Video player had error com.google.android.exoplayer2.ExoPlaybackException: Source error, null, null).

I would appreciate some guidance on how should i perform this or even just if the library allows me to do it.
Thank you very much in advance.

**Comment by jhomlala** [comment]
You can't use localhost in your app in mobile device: https://stackoverflow.com/questions/4779963/how-can-i-access-my-localhost-from-my-android-device

**Reply by ferangmor** [reply]
Thank you very much for your reply, I saw the problem, but is still not working. I am actually calling to those endpoints with ip = 10.0.2.2 and they work perfectly. I am using those to get some files stored in the same database and they get downloaded in my flutter app.
`BetterPlayerDataSource betterPlayerDataSource = BetterPlayerDataSource(
      BetterPlayerDataSourceType.network,
      "http://10.0.2.2:8069/downloadVideoFile/21/4",
      headers: mapa
      );`
Now i am getting this error:
`Unhandled Exception: PlatformException(VideoError, Video player had error com.google.android.exoplayer2.ExoPlaybackException: MediaCodecAudioRenderer error, index=1, format=Format(1, null, null, audio/mp4a-latm, mp4a.40.2, -1, und, [-1, -1, -1.0], [2, 44100]), format_supported=YES, null, null)`

**Reply by jhomlala** [reply]
What's the format of video that you're serving in localhost? You can try apply video format in source configuration in order to help Better Player detect format of video.

**Reply by ferangmor** [reply]
Check:
This is a capture of the Postman call: [Gyazo link](https://gyazo.com/88ce87f14c8245fe160251da8f64c32c)
As you can i get an http response and a video of the type video/mp4.

The server side is written in python, i get the hexadecimal stream of the video from the database, decode it and send it directly, dunno if i should make any inner steps before that.
This is a capture of the endpoint: [gyazo link](https://gyazo.com/823a7a3d5b261a8fe5f460058ccbd58e)

How could i help BetterPlayer to detect the type?

**Reply by jhomlala** [reply]
Use videoFormat parameter from BetterPlayerDataSource: 

```dart
...
 videoFormat: BetterPlayerVideoFormat.hls,
...
```

Can you check with some other players (not in flutter but maybe some desktop) if your data source works correctly? For me it's missconfigured.
